registration/registration_test.go

```diff
@@ -393,17 +446,70 @@ func Test_CheckCertificate(t *testing.T) {
 				Data:      data,
 				Signature: ed25519.Sign(private, data),
 			}
-			_, _, err = r.Submit(context.Background(), challenge, nodeID, 0, r.PowParams(), certificate, time.Time{})
+			_, _, err = r.Submit(
+				context.Background(),
+				challenge,
+				pubKeyHint, nodeID, 0, r.PowParams(), certificate, time.Time{})
 			require.ErrorIs(t, err, registration.ErrInvalidCertificate)
 		})
+
+		t.Run("valid certificate (no public key hint, use main pubkey as default)", func(t *testing.T) {
+			_, _, err = r.Submit(
+				context.Background(),
+				challenge,
+				nil, nodeID, 0, r.PowParams(), defaultValidCert, time.Time{})
+			require.NoError(t, err)
+		})
+
+		t.Run("valid certificate (invalid public key hint)", func(t *testing.T) {
+			invalidHint := []byte{1, 2, 3, 4, 5}
+
+			_, _, err = r.Submit(
+				context.Background(), challenge,
+				invalidHint, nodeID, 0, r.PowParams(), defaultValidCert, time.Time{})
+			require.ErrorIs(t, err, registration.ErrInvalidCertificate)
+		})
+
+		t.Run("valid certificate (with loaded trusted keys)", func(t *testing.T) {
+			_, _, err = r.Submit(
+				context.Background(), challenge,
+				pubKeyHint, nodeID, 0, r.PowParams(), defaultValidCert, time.Time{})
+			require.NoError(t, err)
+
+			key := registration.Base64Enc{}
+			err = key.UnmarshalFlag(string(trustedKey))
+			require.NoError(t, err)
+
+			trustedKeyHint := key.Bytes()[:shared.CertPubkeyHintSize]
```

Comment:
With my suggestion above this just becomes:

```suggestion
			trustedKeyHint := trustedKey[:shared.CertPubkeyHintSize]
```

This test also depends on the execution of another subtest ("certification check enabled"). If the other test is executed first, it will read `valid_key.key` from disk. If the other test is not executed first, that file won't exist and no keys are read from disk.